crypto/rand/randfile.c

```diff
@@ -286,8 +286,22 @@ const char *RAND_file_name(char *buf, size_t size)
         if (OPENSSL_strlcpy(buf, s, size) >= size)
             return NULL;
     } else {
+#ifdef OPENSSL_SYS_WINDOWS
+        /*
+         * We use the same env variables as GetTempFile() - but that function
+         * uses TCHARs, but getenv() gives us chars so its easier to do it this
+         * way
+         */
+        if ((s = getenv("TMP")) == NULL
+            && (s = getenv("TEMP")) == NULL
+            && (s = getenv("USERPROFILE")) == NULL
+            && (s = getenv("SYSTEMROOT")) == NULL) {
+            s = getenv("HOME");
```

Comment:
Thinking about it semantically, is the temporary path something we should use at all?  After all, we don't use /tmp on Unix or anywhere else.  Why on Windows?  So, I'd say that only the following variables should be looked into, in this order:

HOME
USERPROFILE
SYSTEMROOT